_posts/2020-04-11-native-mobile-to-flutter.markdown

```diff
@@ -0,0 +1,114 @@
+---
+layout: post
+title: "Looking Back on Our Migration from Native Mobile to Flutter."
+date: 2020-04-11
+categories: [mobile, flutter]
+author: Bobby Manuel 
+---
+
+Over the last six months, the Ginger mobile team has been busy rewriting our native iOS and Android mobile apps into a shiny new cross-platform Flutter App!  The app has been live in production since February and we are excited to share the incredible gains we have achieved.
+
+### The results are in!
+
+- 2 - 3x improvement in development velocity after the switch.
+- 98% shared code across both platforms.
+- Reduction from 140,000 lines of code to 31,000 lines of code.  This is despite launching with a brand new online/offline content system that did not exist in the native apps.
+- The application polish and user experience is on par with our native iOS application and easily superior to our native android application.
+- Most importantly, we experienced no Flutter related bugs or unpredictable behaviors after the launch.
+
+### How long did this take? 
+
+- Project spanned Aug 2019 - Dec 2019 
+- Initial release to Android users in February 2020.
+- Release to all users Android + iOS March 2020.
+
+### What problems were we hoping to solve?
+
```

Comment:
From my perspective, it was a combination of:

1) economics: we're too small to have dedicated native iOS and Android apps; thankfully technology has evolved to make a single app/codebase work well for both platforms
2) social dynamics - we wanted to have a single mobile app team.  too much gets wasted in the industry because of the rift between iOS/Android teams.  having a single team greatly reduces the social complexity of building an app, so we can move faster and have a better unified product experience

Reply:
Agreed - i'm stealing these points - thanks!